src/mappings/nfts/buy.ts

```diff
@@ -15,18 +15,20 @@ export async function handleTokenBuy(context: Context): Promise<void> {
   debug(OPERATION, event, true)
 
   const id = createTokenId(event.collectionId, event.sn)
-  const entity = await getWith(context.store, NE, id, { collection: true })
+  const entity = await getWith<NE>(context.store, NE, id, { collection: true })
+  const collection = await getWith<CE>(context.store, CE, event.collectionId)
 
-  const originalPrice = entity.price
+  const originalPrice = event.price
   const originalOwner = entity.currentOwner ?? undefined
 
   entity.price = BigInt(0)
   entity.currentOwner = event.caller
   entity.updatedAt = event.timestamp
+  
   if (originalPrice) {
-    entity.collection.volume += originalPrice
-    if (originalPrice > entity.collection.highestSale) {
-      entity.collection.highestSale = originalPrice
+    collection.volume += BigInt(originalPrice)
+    if (BigInt(originalPrice) > BigInt(collection.highestSale)) {
+      collection.highestSale = BigInt(originalPrice)
```

Comment:
not sur if the `BigInt(..)` here are needed

Reply:
Most possibly not